test/functional/blocknotify.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test blocknotify.
+
+Verify that a bitcoind node calls the given command for each block
+"""
+
+import os
+import time
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class BlockNotifyTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+
+        self.setup_clean_chain = False
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.path = os.path.join(self.options.tmpdir, 'blocks.txt')
+
+        # the command will append the notified block hash to file
+        self.extra_args = [['-blocknotify=echo %%s >> %s' % self.path]]
+        super().setup_network()
+
+    def run_test(self):
+        block_count = 10
+        blocks = self.nodes[0].generate(block_count)
+
+        # wait for expected data length
```

Comment:
A few comments on this loop:

- I'd make the sleep longer (say 0.1 seconds). I'm also not sure you need the back-off. Just a `time.sleep(0.1)` should be fine.
- The total time allowed for the file to be written is 0.45 seconds. That needs to be longer (up to 10 seconds) since Travis can block i/o for many seconds at a time. See #10072.
- Please make the comment clear about the total time allowed for the loop.